source/js/components/currency-selector.js

```diff
@@ -171,20 +173,48 @@ class CurrencySelect {
   }
 
   bindOtherAmountEvents() {
-    // Other amount vars
-    this.otherAmountInput = document.querySelectorAll("[data-other-amount]");
-    this.otherAmountLabel = document.querySelectorAll("[data-currency]");
-    this.otherAmountRadio = document.querySelectorAll(
+    var otherAmountInputs = document.querySelectorAll("[data-other-amount]");
+    var otherAmountRadios = document.querySelectorAll(
       "[data-other-amount-radio]"
     );
-    for (var i = 0; i < this.otherAmountInput.length; i++) {
-      this.otherAmountInput[i].addEventListener("click", event =>
-        this.selectRadio(event)
-      );
-      this.otherAmountInput[i].addEventListener("change", event =>
-        this.updateValue(event)
+    otherAmountInputs.forEach(input => {
+      input.addEventListener(
+        "click",
+        event =>
+          (input.parentNode.querySelector(
+            "[data-other-amount-radio]"
+          ).checked = true)
       );
-    }
+      input.addEventListener("change", event => this.updateValue(input));
+      input.addEventListener("focusin", event => this.updateValue(input));
+    });
+
+    otherAmountRadios.forEach(input => {
+      input.addEventListener("change", event => {
+        if (input.checked) {
+          this.updateValue(
+            input.parentNode.querySelector("[data-other-amount]")
+          );
+        }
+      });
+    });
+
+    // Because we are using minimum amount validation on the other amount,
+    // we need to make sure that this amount is emptied if the user
+    // selects a preset amount. Otherwise the browser will fail to validate the form.
+    document
+      .querySelectorAll(
+        ".donation-amount__radio:not([data-other-amount-radio])"
+      )
+      .forEach(input => {
+        input.addEventListener("change", e => {
+          if (input.checked) {
+            otherAmountInputs.forEach(
+              input => (input.value = input.checkValidity() ? input.value : "")
```

Comment:
was this supposed to be `===`? Because as a tertiary based on an assignment return this is very scary =)